biggus/__init__.py

```diff
@@ -2287,6 +2346,103 @@ def _sliced_shape(shape, keys):
     return sliced_shape
 
 
+class broadcast(object):
+    def __init__(self, array1, array2):
+        self._a1 = array1
+        self._a2 = array2
+        #: The broadcast shape.
+        self.shape = broadcast.compute_broadcast_shape(array1.shape,
+                                                       array2.shape)
+
+    @staticmethod
+    def compute_broadcast_shape(shape1, shape2):
```

Comment:
Hmm. I guess I lead you astray. `as_strided` does work a little better, but it still runs into trouble for truly gigantic arrays:
```
In [20]: as_strided(np.array(0), (100000,) * 4, (0,) * 4)
ValueError: array is too big.
```

On the other hand, a virtual array that big would be measured in exabytes.